### PR TITLE
Do not throw SpanContextParseException upon extraction.

### PR DESCRIPTION
**Pull request by carlosalberto**
Addresses #36 

Observe that, now, `extract()` methods are expected to return `Span.INVALID` in case the context could not be parsed.

### CODE REVIEW

**Comment by yurishkuro** [comment]
Is this INVALID context still OK to use as a parent?

The use case we have in Jaeger is sending a curl request either `jaeger-debug-id` header or `jaeger-baggage` header, in order to tag a request in some way. There's no actual trace id in the request, so Jaeger returns a SpanContext that is technically "invalid", but it still captures those two headers so that when it's used as a parent to start a span (which starts a new trace), the span inherits these extra attributes.

**Comment by carlosalberto** [comment]
hey hey @yurishkuro 

In a late call yesterday it was mentioned that the `Tracer` has to be aware of this **specific** value (`SpanContext.INVALID`, that is) and then the proper logic will be applied (meaning: no actual parent/context).

**Comment by yurishkuro** [comment]
That won't solve the use case I described if we make it a requirement to return INVALID constant. Since OpenTracing leaves this up to the tracer, the implementations can use internal mechanisms to mark the span context as invalid.

**Comment by yurishkuro** [comment]
In other words, if we really want to expose the knowledge of invalid to the user, it's better to have a method on it rather than a public constant (my preference is not to expose it at all).

**Comment by carlosalberto** [comment]
Or otherwise return `null` as we do in OT, @yurishkuro ?

**Comment by yurishkuro** [comment]
I would not be opposed to a requirement to always return non-null value (implementation can always use a constant if it wants). What I am opposed to is defining `SpanContext.INVALID` in the API and **requiring** to return it when no trace context is found, since it locks out the possibility of passing other implementation-specific values that may indicate "partial" context.

**Comment by bogdandrutu** [comment]
+1 to always return not null and require instrumentation code to pass the returned value to the Builder so that implementation may decide to use "partial" span-context data.

SpanContext.INVALID can be a suggestion to be returned if data cannot be parsed.

**Comment by tedsuo** [comment]
This PR looks good. We also want the following next steps:
* Create a new PR to remove `SpanContext.INVALID`
* Never return `null`, and allow underlying implementation to decide how to handle partial data.

File: api/src/main/java/openconsensus/trace/propagation/BinaryFormat.java [hunk 1]
```diff
@@ -80,10 +76,10 @@
    * Parses the {@link SpanContext} from a byte array using the binary format.
    *
    * @param bytes a binary encoded buffer from which the {@code SpanContext} will be parsed.
-   * @return the parsed {@code SpanContext}.
+   * @return the parsed {@code SpanContext} or {@code SpanContext.INVALID} if it could not be
```
Comment (bogdandrutu) [inline api/src/main/java/openconsensus/trace/propagation/BinaryFormat.java]: We said that we don't want to mention SpanContext.INVALID here.

Comment (carlosalberto) [reply api/src/main/java/openconsensus/trace/propagation/BinaryFormat.java]: Updated this ;)

**Pull request closed without merging**